crates/uv/tests/it/pip_list.rs

```diff
@@ -176,6 +176,49 @@ fn list_outdated_json() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn list_outdated_find_links() {
+    let context = TestContext::new("3.12");
+
+    let links_dir = context.workspace_root.join("scripts/links");
+
+    uv_snapshot!(context.filters(), context.pip_install()
+        .arg("validation==1.0.0")
+        .arg("--find-links")
+        .arg(&links_dir)
+        .arg("--no-index"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 1 package in [TIME]
+    Prepared 1 package in [TIME]
+    Installed 1 package in [TIME]
+     + validation==1.0.0
+    "###
+    );
+
+    uv_snapshot!(context.filters(), context.pip_list()
+        .arg("--outdated")
+        .arg("--find-links")
+        .arg(&links_dir)
+        .arg("--no-index"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Package    Version Latest Type
+    ---------- ------- ------ -----
+    validation 1.0.0   3.0.0  wheel
+
+    ----- stderr -----
+    warning: validation-1.0.0-py3-none-any.whl is missing an upload date, but user provided: global: 2024-03-25T00:00:00Z
```

Comment:
We can run this test without the exclude newer env var, those warnings are misleading.